File: AllSamples/Samples/Samples_Source/indigo/remoteobjects/remoteobjects.helloremoteinterface_vb/server/remotableobject.vb
Imports System
Namespace Microsoft.Samples.MessageBus.Quickstarts.RemoteObjects
    Public Class RemotableObject
        Implements IRemoteInterface

        Private clientCount As Integer = 0

        Public Sub New()
            Console.WriteLine("Object {0} has been created.", Me.GetHashCode().ToString())
        End Sub
        Public Function Hello(ByVal name As String) As String Implements IRemoteInterface.Hello
            clientCount = clientCount + 1
            Console.WriteLine("{0} has said Hello. This instance has been called {1} times.", name, clientCount.ToString())
            Return String.Format("Hello, {0}. This is RemotableObject instance ""{1}""", name, Me.GetHashCode().ToString())
        End Function
        Protected Overrides Sub Finalize()
            Console.WriteLine("Object {0} is being torn down.", Me.GetHashCode().ToString())
        End Sub
    End Class
End Namespace

File: AllSamples/Samples/Samples_Source/indigo/remoteobjects/remoteobjects.hellogetremoteobject_cs/server/remotableobject.cs
using System; 

namespace Microsoft.Samples.MessageBus.Quickstarts.RemoteObjects
{
    public class RemotableObject : MarshalByRefObject
    {
        private int clientCount = 0;

        public RemotableObject()
        {
            Console.WriteLine("Object {0} has been created.", this.GetHashCode().ToString());
        }

        public string Hello(string name)
        {    
            ++clientCount;
            Console.WriteLine("{0} has said Hello. This instance has been called {1} times.", name, clientCount.ToString());
            return String.Format("Hello, {0}. This is RemotableObject instance \"{1}\"", name, this.GetHashCode().ToString());
        }
    
        ~RemotableObject()
        {
            Console.WriteLine("Object {0} is being torn down.", this.GetHashCode().ToString());
        }
    }
}
